### ISSUE TITLE
Popup closes automatically when clicked

### ISSUE THREAD

**Issue by Clafouti** [YUzhva/react-leaflet-markercluster#22]
Hi,

When I use the onMarkerClick event like this : 

```
  onMarkerClusterClick(marker) {
    this.setState({
      currentMarker: this.markersDirectAccess[marker.options.icon.options.internalId]
    });
  }
```

The popup attached to the markers closes automatically. But when I do the same with a single marker (using react-leaflet) it stays open.
To be honest, I don't know where to look to resolve this issue... Any help is appreciated.

**Comment by YUzhva** [comment]
@Clafouti looks like the issue because of setState.

I'll try to reproduce this issue tomorrow and will answer/publish fix asap.

**Comment by Clafouti** [comment]
Nice. I can share more code If needed, just ping me.

**Comment by YUzhva** [comment]
Just one question: are you sending markers and array through props like:
```javascript
<MarkerClusterGroup
    markers={markers}
  />
```
Or you are using `Marker` from `react-leaflet`
```javascript
<MarkerClusterGroup wrapperOptions={{enableDefaultStyle: true}}>
    <Marker position={[49.8397, 24.0297]} />
    <Marker position={[52.2297, 21.0122]} />
    <Marker position={[51.5074, -0.0901]} />
  </MarkerClusterGroup>
```
:question:

**Comment by Clafouti** [comment]
First version : 

```
<Overlay name='Clustered markers'>
  <LayerGroup>
    <MarkerClusterGroup
      markers={this.state.markers}
      wrapperOptions={{enableDefaultStyle: true}}
      onMarkerClick={this.onMarkerClusterClick}
    />
  </LayerGroup>
</Overlay>

```

**Comment by Clafouti** [comment]
@YUzhva Any news about that ?

**Comment by YUzhva** [comment]
@Clafouti 

Fixed in: [release] 1.1.6: Fix update markers on a map only when new props diff. from prev. (#27)

P.S: Could you, please try and close that issue if there everything ok?

**Comment by Clafouti** [comment]
Works perfectly now ! Thank you.
